aws_lambda_powertools/logging/types.py

```diff
@@ -0,0 +1,10 @@
+from typing import Any, Dict, Iterable, Union
+
+from typing_extensions import TypedDict
+
+class TypedLog(TypedDict, total=False):
```

Comment:
Two suggestions here:

* Rename to a more explicit name like `PowertoolsLogRecord` - this makes it easier to quickly understand what this is about
* Remove `total=False` as that removes our guarantee that these keys will be present. There's another way you can be more assertive about what's required vs what's not required - sending a snippet for you